lib/node/actions/http_client.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+
+import 'package:js/js.dart' as js;
+
+@JS()
+@js.staticInterop
+class HttpClient {
+  external factory HttpClient(
```

Comment:
Up to you if you want to change the parameter types here to JS types.

Reply:
sgtm - updates to use JSString and JSArray